bpf/lib/nat.h

```diff
@@ -445,18 +445,25 @@ static __always_inline int snat_v4_rewrite_ingress(struct __ctx_buff *ctx,
 			return DROP_CSUM_L4;
 #endif  /* ENABLE_SCTP */
 		case IPPROTO_ICMP: {
+			__u8 type = 0;
 			__be32 from, to;
 
-			if (ctx_store_bytes(ctx, off +
-					    offsetof(struct icmphdr, un.echo.id),
-					    &state->to_dport,
-					    sizeof(state->to_dport), 0) < 0)
-				return DROP_WRITE_ERROR;
-			from = tuple->dport;
-			to = state->to_dport;
-			flags = 0; /* ICMPv4 has no pseudo-header */
-			sum_l4 = csum_diff(&from, 4, &to, 4, 0);
-			csum.offset = offsetof(struct icmphdr, checksum);
+			if (ctx_load_bytes(ctx, off +
+					   offsetof(struct icmphdr, type),
+					   &type, 1) < 0)
+				return DROP_INVALID;
+			if (type == ICMP_ECHO || type == ICMP_ECHOREPLY) {
+				if (ctx_store_bytes(ctx, off +
+						    offsetof(struct icmphdr, un.echo.id),
+						    &state->to_dport,
+						    sizeof(state->to_dport), 0) < 0)
+					return DROP_WRITE_ERROR;
+				from = tuple->dport;
+				to = state->to_dport;
+				flags = 0; /* ICMPv4 has no pseudo-header */
+				sum_l4 = csum_diff(&from, 4, &to, 4, 0);
+				csum.offset = offsetof(struct icmphdr, checksum);
+			}
```

Comment:
Unless I'm missing something, this is a functional change. What's the impact for users? Does it change how the echo/echo-reply messages are treated? What will be the visible change?

Reply:
Since we do currently only support echo/echo-reply no changes are expected from an user POV.

Reply:
> we do currently only support echo/echo-reply

What does this mean exactly? Do you mean in the documentation? If someone tries to send something other than echo/echo-reply what happens? Or do you mean that the information collected here is dropped later on if the packet is not an ICMP echo or echo-reply (in which case this is *not* a functional change)?

Reply:
Oh, yes yes there are no functional changes today as the packet will be dropped before to join this function [0]. The point will be once we will support more types.

[0] https://github.com/cilium/cilium/pull/22004/files#diff-ef710180f228e8f6bbad249e2670dbc42a54f45da3bbc4c5aebbb1f3865979e9R779

Reply:
Ok, I see.

Could you update the commit description to make that clear? Right now the commit title sounds like this commit is limiting the rewrite of ICMP messages on ingress when it's not changing anything.

I think you can also merge the last two commits in one, though it doesn't matter much.